### PR TITLE
#730 `x/act` rename rule to template

### PR DESCRIPTION
**Pull request by backsapc**
https://github.com/warden-protocol/wardenprotocol/issues/730

## Summary by CodeRabbit

- **New Features**
	- Introduced Votes and Approve/Reject expressions for Actions.
	- Enhanced MsgAddAction with expected expressions.

- **Terminology Updates**
	- Renamed "Rule" to "Template" throughout the application for improved clarity. 

These changes aim to enhance the expressiveness and functionality of actions within the system, making it easier for users to understand and interact with the features.

### CODE REVIEW

**Comment by artur-abliazimov** [comment]
Check the `/warden/x/act/types/v1beta1/expression.go` + `/warden/x/act/types/v1beta1/registry.go`.

**Comment by artur-abliazimov** [comment]
`query_rules.go` and other related to `rule` files must be renamed + I see some `Rule` mentions there.
image

File: warden/x/warden/keeper/msg_server_update_key.go [hunk 1]
```diff
@@ -17,11 +17,25 @@ func (k msgServer) UpdateKey(ctx context.Context, msg *types.MsgUpdateKey) (*typ
...
+	}
+
+	if key.RejectTemplateId != msg.RejectTemplateId {
+		if err = k.actKeeper.IsValidTemplate(ctx, msg.RejectTemplateId); err != nil {
+			return nil, err
+		}
+		key.RejectTemplateId = msg.RejectTemplateId
 	}
 
 	if key.ApproveRuleId != msg.ApproveRuleId {
```
Comment (artur-abliazimov) [inline warden/x/warden/keeper/msg_server_update_key.go]: `Rule` is still here.

File: warden/x/act/keeper/actions.go [hunk 2]
```diff
@@ -103,6 +104,7 @@ func (k Keeper) TryRejectVotedAction(ctx context.Context, act *types.Action) err
 // result in the database.
 func (k Keeper) TryExecuteAction(ctx context.Context, act *types.Action) error {
 	ready, err := k.checkActionReady(ctx, *act)
+	log.Printf("\nready %v\n", ready)
```
Comment (artur-abliazimov) [inline warden/x/act/keeper/actions.go]: It must be removed

**Comment by artur-abliazimov** [comment]
Do we also need to fix `rule` in the following files?

 - /tests/testdata/snapshot-many-users/config/genesis.json
 - /tests/testdata/snapshot-keychain/config/genesis.json
 - /wardenprotocol/tests/testdata/snapshot-base/config/genesis.json

**Comment by backsapc** [comment]
@artur-abliazimov great catch! It's not critical, but yes

**Review by artur-abliazimov — APPROVED** [review]

**Pull request merged**